File: Data Structures & Algorithms/anagram-groups/submission-9.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> hm = new HashMap<>();
        for(String s : strs) {
            StringBuilder hash = new StringBuilder();
            int[] freq = new int[26];
            for(char c : s.toCharArray()) {
                freq[c-'a']++;
            }
            for(int i=0; i<26; i++) {
                hash.append(freq[i]);
                hash.append('#'); 
            }
            List<String> tmp = hm.getOrDefault(hash.toString(), new ArrayList<>());
            tmp.add(s);
            hm.put(hash.toString(), tmp);
        }
        List<List<String>> ans = new ArrayList<>();
        for(List<String> l : hm.values())
            ans.add(l);
        return ans;
    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-9.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        dp = [[False]*n for _ in range(n)]
        ans = [0]*2
        maxlen = 1
        for i in range(n):
            dp[i][i] = True
            if i!=n-1 and s[i]==s[i+1]:
                dp[i][i+1] = True
                ans[0], ans[1] = i, i+1
                maxlen = 2
        for i in range(n-3, -1, -1):
            for j in range(i+2, n):
                if s[i]==s[j]:
                    dp[i][j] = dp[i+1][j-1]
                    if dp[i][j] and j-i+1 > maxlen:
                        ans[0], ans[1] = i, j
                        maxlen = j-i+1
        return s[ans[0]:ans[1]+1]

            